dll/win32/shell32/CCopyAsPathMenu.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * PROJECT:     ReactOS shell32
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Copy as Path Menu implementation
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ *              Copyright 2024 Thamatip Chitpong <thamatip.chitpong@reactos.org>
+ */
+
+#include "precomp.h"
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+CCopyAsPathMenu::CCopyAsPathMenu() :
+    m_idCmdFirst(0),
+    m_idCmdAction(-1)
```

Comment:
Why are you storing these when you don't have to?

Just `#define IDC_COPYASPATH 0`

Then do
- `mii.wID = idCmdFirst + IDC_COPYASPATH;`
- `if (IS_INTRESOURCE(lpcmi->lpVerb) && LOWORD(lpcmi->lpVerb) == IDC_COPYASPATH)`

The host of IContextMenu is responsible for mapping the id so you don't have to keep track of idCmdFirst.

Reply:
Done.